### ISSUE TITLE
--split-output and --split-input behave differently with escape sequences

### ISSUE THREAD

**Issue by maxrothman** [Russell91/pythonpy#82]
Reproduce:
```bash
# $'\t' is how you get a literal tab character in bash
$ seq 1 10 | xargs -I% echo '%'$'\t''foo'
1	foo
2	foo
3	foo
4	foo
5	foo
6	foo
7	foo
8	foo
9	foo
10	foo

$ seq 1 10 | xargs -I% echo '%'$'\t''foo' | py --si '\t' -x 'x'
['1', 'foo']
['2', 'foo']
['3', 'foo']
['4', 'foo']
['5', 'foo']
['6', 'foo']
['7', 'foo']
['8', 'foo']
['9', 'foo']
['10', 'foo']

$ seq 1 10 | xargs -I% echo '%'$'\t''foo' | py --si '\t' --so '\t' -x 'x'
1\tfoo
2\tfoo
3\tfoo
4\tfoo
5\tfoo
6\tfoo
7\tfoo
8\tfoo
9\tfoo
10\tfoo
```

As you can see, `py` successfully splits on tabs when used with `--si '\t'`, but when used with `--so '\t'`, interprets it as the characters `['\', 't']`

__Expected behavior:__ both `--si` and `--so` interpret escape sequences in a consistent manner

__Actual behavior:__  `--si` converts escape sequences into the characters they represent, while `--so` does not.

This is likely due to the use of `re.split()` [here](https://github.com/Russell91/pythonpy/blob/master/pythonpy/__main__.py#L115):
```python
>>> print 'foo\tbar'
foo	bar
>>> print r'foo\tbar'
foo\tbar
>>> re.split(r'\t', 'foo\tbar')
['foo', 'bar']
>>> re.split('\t', 'foo\tbar')
['foo', 'bar']
```
This behavior is the same in Python 2 and 3.

I considered making a pull request to fix this, but I wasn't sure what direction you wanted to go in. If you prefer that escape sequences are interpreted by default, then using `args.output_delimiter.decode('string-escape')` would fix the issue<sup>1</sup>. If you prefer escape sequences to **not** be interpreted by default, either the backslashes in the `--split-input` arg would have to be escaped, or the current behavior would be moved under a new option (e.g. `--rsplit-input`) and `--split-input` would be switched to simply use `str.split()`<sup>2</sup>.

<br>
<br>

<sup>1</sup>In Python 3, `string-escape` was renamed to `unicode-escape`, and the `.decode()` method is only available on `bytes` objects, so a little version checking would have to be done.

<sup>2</sup>FWIW, I think the latter is a better idea

**Comment by vladsaveliev** [comment]
+1 to this. Would really appreciate if this options were finalised and exposed!

**Comment by Russell91** [comment]
@maxrothman I agree that this is a problem. I'm a bit concerned about changing backwards compatible behavior though. What would you think about creating a new version that changes a few issues, e.g. including dictionary printing only the keys. Could be put up on `pip install pythonpy2`.
